pkg/ipcache/ipcache.go

```diff
@@ -350,6 +356,11 @@ func (ipc *IPCache) deleteLocked(ip string) {
 		return
 	}
 
+	if cachedIdentity.Source != source {
+		scopedLog.Debugf("Skipping delete of identity from source %s", source)
+		return
+	}
+
```

Comment:
Would it ease backporting if the bug fix would be separated to it's own commit?

Reply:
👍 Factored out into https://github.com/cilium/cilium/pull/6475